### ISSUE TITLE
make job_data() messages more clear when job is pending

### ISSUE THREAD

**Issue by nickreich** [reichlab/forecast-repository#235]
See below for when the first time I ran the job_data() to try to get a job it didn't work and then it worked. i think this is because it was still pending/in queue?

```
> job_url <- zoltr::submit_query(zoltar_connection, project_url, zoltar_query)
> ## extract query data
> tmp <- job_data(zoltar_connection, job_url)
get_resource(): GET: https://www.zoltardata.com/api/job/22721/data/
Error in get_resource(zoltar_connection, data_url) : 
  Internal Server Error (HTTP 500).
> ## extract query data
> tmp <- job_data(zoltar_connection, job_url)
get_resource(): GET: https://www.zoltardata.com/api/job/22721/data/
Parsed with column specification:
cols(
  model = col_character(),
  timezero = col_date(format = ""),
  season = col_character(),
  unit = col_character(),
  target = col_character(),
  class = col_character(),
  value = col_double(),
  cat = col_logical(),
  prob = col_logical(),
  sample = col_logical(),
  quantile = col_logical(),
  family = col_logical(),
  param1 = col_logical(),
  param2 = col_logical(),
  param3 = col_logical()
)
```

**Comment by matthewcornell** [comment]
This happens when you don't call `busy_poll_job()` before calling `job_data()`. What's happening is that the job is not done running the query and so has not yet saved the results to the underlying S3 file (it's this file that is ultimately returned). So the file is not there when you call `job_data()`, which is a failure condition. As of 7/31 I implemented #240 [download_job_data() sometimes returns a 500 Internal Server Error #240] in which Zoltar returns a 404 Not Found error instead of a 500 (server) one. This is not perfect (it's still an error) but the error message gives a hint.

I will update the docs to capture this.

**Comment by nickreich** [comment]
I think this is basically ok now that we've implemented the `do_zoltar_query()` function. maybe we could close this issue until we see more errors while using that function?
